mood-analysis.rb

```diff
@@ -14,11 +15,35 @@ def analyze_mood(words)
       sad += 1
     end
   end
+  # return words
   return ":-)" if happy > sad
   return ":-(" if happy < sad
   return ":-|"
 end
 
+def strip_punctuation(string_of_words)
+  string_of_words.gsub(/[^a-z0-9\s]/i, '')
+end
+
+def happy_days(text_array)
+  days = 0
+  happydays = 0
+  text_array.each do |entry|
+    result = analyze_mood(entry)
+    days += 1
+    happydays += 1 if result == ":-)"
+    break if happydays == 3
+  end
+  happydays < 3 ? eval = "Oh no! Out of #{days} days there were #{happydays} happy days." : eval = "It takes #{days} days for 3 happy days to occur."
+  puts eval
+end
```

Comment:
This is good! The only thing I don't like is that the method is returning a string instead of a fixnum accounting for how many days that it takes for 3 happy days. This means that we have less flexibility in using this information in other ways.